internal/services/report_processor.go

```diff
@@ -172,13 +173,13 @@ func ProcessReport(msg *kafka.Message, _ *kafka.Consumer) {
 							IntervalEnd:   interval_end_time,
 							UsageMetrics:  container_usage_metrics,
 						}
-						if err := workload_metric.CreateWorkloadMetrics(); err != nil {
-							log.Errorf("unable to add record to workload_metrics table: %v. Error: %v", workload_metric, err)
-							continue
-						}
+						workload_metric_arr = append(workload_metric_arr, workload_metric)
```

Comment:
When we are unable to parse `interval_start_time` or `interval_end_time` then we skip the iteration and go to next one. However for `container_usage_metrics` I see we dont have `continue`,
~~~
container_usage_metrics, err := json.Marshal(container.Metrics)
if err != nil {
  log.Errorf("Unable to marshal container usage data: %v", err)
}
~~~

Considering the fact that nothing in `workload_metrics` is nullable, we should also use continue in above if block? so this way we don't error out while inserting or updating in database later in BatchInsertWorkloadMetrics? and if I understand correctly if one of the row is missing `workload_metrics` then it won't process whole batch?

P.S its not related to this pull request, however as we are switching to array I think this can affect data insertion or update if there are some rows which are missing the fields

Reply:
I agree. I will add `continue` for `container_usage_metrics` block.

Reply:
Done